### PR TITLE
Replace abstract test kernel with customized test kernel

### PR DESCRIPTION
**Pull request by jrushlow**
The abstract test kernel was dependent on the `MicroKernelTrait` provided by the framework bundle. Between changes within the trait and outside dependency requirements, the abstraction became to cumbersome. 

This PR replaces the abstraction with a new `VerifyEmailTestKernel::class` that is strictly for use in the higher level tests cases. Some aspects are a bit more verbose, but it cuts out most of the boiler plate needed for the tests. Most importantly, it is a bit more immune to outside package changes.

### CODE REVIEW

File: tests/VerifyEmailTestKernel.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                ]
+            );
+
+            $container->addObjectResource($this);
+            $container->register('kernel', static::class)
+                ->addTag('controller.service_arguments')
+                ->setAutoconfigured(true)
+                ->setSynthetic(true)
+                ->setPublic(true)
+            ;
```
Comment (weaverryan) [inline tests/VerifyEmailTestKernel.php:83]: I don't think you need this section - almost definitely not the `->addTag()` part, but maybe also ALL of these lines can be removed. We're just not making use of them.

File: tests/VerifyEmailTestKernel.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+            $kernelDefinition = $container->getDefinition('kernel');
+            $kernelDefinition->addTag('routing.route_loader');
```
Comment (weaverryan) [inline tests/VerifyEmailTestKernel.php:86]: I think the same down here... but I can't exactly remember what this does...

File: tests/VerifyEmailTestKernel.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ */
+class VerifyEmailTestKernel extends Kernel
+{
+    private $builder;
+    private $routes;
+    private $extraBundles;
+
+    /**
+     * @param array $routes  Routes to be added to the container e.g. ['name' => 'path']
+     * @param array $bundles Array of bundles to be registered e.g. [new Bundle()]
```
Comment (weaverryan) [inline tests/VerifyEmailTestKernel.php:33]: @param `BundleInterface[]`

Comment (jrushlow) [reply tests/VerifyEmailTestKernel.php]: done

**Pull request merged**